test/gtest/ucs/test_conn_match.cc

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include <common/test.h>
+
+#include <ucs/datastruct/conn_match.h>
+#include <ucs/sys/sys.h>
+
+#include <vector>
+
+class test_conn_match : public ucs::test {
+public:
+    typedef struct {
+        ucs_conn_match_elem_t       elem;
+        ucs_conn_match_queue_type_t queue_type;
+        std::string                 dest_address;
+        ucs_conn_sn_t               conn_sn;
+    } conn_elem_t;
+
+    test_conn_match() {
+        m_address_length = 0;
+    }
+
+private:
+    void conn_match_init(size_t address_length) {
+        ucs_conn_match_ops_t conn_match_ops;
+
+        conn_match_ops.get_address = get_address;
+        conn_match_ops.get_conn_sn = get_conn_sn;
+        conn_match_ops.address_str = address_str;
+
+        ucs_conn_match_init(&m_conn_match_ctx, address_length,
+                            &conn_match_ops);
+        m_address_length = address_length;
+    }
+
+protected:
+    static const void* get_address(const ucs_conn_match_elem_t *conn_match) {
+        return ucs_container_of(conn_match, conn_elem_t,
+                                elem)->dest_address.c_str();
+    }
+
+    static ucs_conn_sn_t get_conn_sn(const ucs_conn_match_elem_t *conn_match) {
+        return ucs_container_of(conn_match, conn_elem_t,
+                                elem)->conn_sn;
+    }
+
+    static const char *address_str(const void *address, char *str,
+                                   size_t max_size) {
+        return ucs_strncpy_safe(str, static_cast<const char*>(address),
+                                ucs_min(m_conn_match_ctx.address_length,
+                                        max_size));
+    }
+
+    void init_new_address_length(size_t address_length) {
+        if (address_length != m_address_length) {
+            ucs_conn_match_cleanup(&m_conn_match_ctx);
+            conn_match_init(address_length);
+        }
+    }
```

Comment:
why not just test with single address length like 64?

Reply:
wanted to test several address lengths to catch potential bugs